### PR TITLE
Config upload

### PR DESCRIPTION
**Pull request by fulibacsi**
Added config endpoint to upload configs by posting. Added a simple 404 page.

### CODE REVIEW

**Comment by zoltanctoth** [comment]
@fulibacsi 
 1. can you add a piece in README about how to use it?
 2. Just to check (I am not there yet with review): Will it break installations with existing config files?

**Comment by fulibacsi** [comment]
> @fulibacsi
> 
> 1. can you add a piece in README about how to use it?
Sure, that's a good idea.
> 2. Just to check (I am not there yet with review): Will it break installations with existing config files?
It is not a breaking change.

File: configs/loader.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if not contains(['workspaces', 'prices'], config):
+        error_msg = json.dumps({'success': False,
+                                'error': 'Missing config info',
+                                'info': config})
+        raise Exception(error_msg)
+
+    req_workspace_keys = ['url', 'id', 'type', 'name', 'token']
+    for workspace in config['workspaces']:
+        if not contains(req_workspace_keys, workspace):
+            error_msg = json.dumps({'success': False,
```
Comment (zoltanctoth) [inline configs/loader.py:36]: Shouldn't we report back which workspace fields were missing?

Comment (fulibacsi) [reply configs/loader.py:43]: Agreed, implemented.

**Comment by fulibacsi** [comment]
Readme updated.

File: README.md [hunk 2]
```diff
@@ -37,6 +37,40 @@ _configs/config.json_
 }
 ```
 
+#### Sending config during run
+
+After starting the UI, config files can still by send in. To submit a new configuration, use the `ui_uri:port/config` REST API endpoint: send the configuration as `json` payload in a `POST` request. Please note that currently the whole configuration must be submitted, the old one will be replaced.
```
Comment (zoltanctoth) [inline README.md:42]: ff
```suggestion
After starting the UI, config files can still be added by HTTP send in. To submit a new configuration, use the `ui_uri:port/config` REST API endpoint: send the configuration as a `json` payload in a `POST` request. Please note that currently the whole configuration must be submitted, the old configuration will be replaced.
```

File: README.md [hunk 3]
```diff
@@ -37,6 +37,40 @@ _configs/config.json_
...
+
+Submission example using python:
+```python
+import requests
+config = { ... }
```
Comment (zoltanctoth) [inline README.md:47]: Please add a dummy config, so people know what is the root tag of the json and what is expected specifically

File: README.md (hunk 2 above)
Comment (fulibacsi) [reply README.md:42]: Fixed.

File: README.md (hunk 3 above)
Comment (fulibacsi) [reply README.md:47]: Added.

File: dac.py [hunk 4]
```diff
@@ -48,11 +50,18 @@ def format_datetime(value):
 
 @functools.lru_cache(maxsize=None)
```
[the remark this answers is not in the document]
Comment (fulibacsi) [reply dac.py:51]: Max size set.

File: dac.py [hunk 5]
```diff
@@ -48,11 +50,18 @@ def format_datetime(value):
...
+    try:
+        if path is None:
+            path = app.config.get('CONFIG_PATH')
+
+        with open(path, 'r') as config_file:
+            settings = json.load(config_file)
+    except Exception:
+        abort(404, {'error': 'Missing',
+                    'type': 'config file',
+                    'message': 'Config file is missing. Upload one using by '
```
[the remark this answers is not in the document]
Comment (fulibacsi) [reply dac.py:62]: Added.

**Pull request merged**